nautobot/dcim/tables/devices.py

```diff
@@ -1411,3 +1412,48 @@ class Meta(BaseTable.Meta):
             "tags",
             "actions",
         )
+
+
+class VirtualDeviceContextTable(StatusTableMixin, BaseTable):
```

Comment:
Is this in conflict with #6209? Should this branch be based under that one?

Reply:
I dont think this is in conflict, I pretty much copied the VirtualDeviceContextTable from #6029 and added the role column to it. But I do think #6029 should be merged first.

Reply:
I think its better your base branch is #6209